Samples/iOS-Swift/iOS-Swift-UITests/BaseUITest.swift

```diff
@@ -46,10 +46,17 @@ extension BaseUITest {
         // Calling activate() and then launch() effectively launches the app twice, interfering with
         // local debugging. Only call activate if there isn't a debugger attached, which is a decent
         // proxy for whether this is running in CI.
-        if !isDebugging() {
+        if !isDebugging() && activateBeforeLaunch {
+            // activate() appears to drop launch args and environment variables, so save them beforehand and reset them before subsequent calls to launch()
+            let launchArguments = app.launchArguments
+            let launchEnvironment = app.launchEnvironment
+
             // App prewarming can sometimes cause simulators to get stuck in UI tests, activating them
             // before launching clears any prewarming state.
             app.activate()
```

Comment:
@noahsmartin I was unable to resolve the UI test failures until adding this facility to avoid the double-launch. I think what was happening is that the files written were wiped on each launch. I didn't see another way to deal with all the implications of an individual launch for these tests, since they're sensitive to every launch of the app, and try to control that exactly to launch each app twice per test method.